### PR TITLE
Make native sln builds opt in rather than always on

### PR DESCRIPTION
**Pull request by jkotalik**
https://github.com/aspnet/AspNetCore/pull/11805 had some side effects that caused the Components sln to break due to depending on the shared framework, which has a native dependency. This reverts back to using an environment variable in StartVs to opt into building native components.

### CODE REVIEW

**Comment by Tratcher** [comment]
Not a fan of having startvs behave differently in different directories. It was more convenient to be able to use startvs in the root and give it the path to an sln.

**Comment by jkotalik** [comment]
Ah good point how that breaks that workflow. I never use that but I understand the concern.

I think we can both agree that the behavior we want is when you launch the IIS/ServerTests/Music store slns, we want to build all native assets as well as managed. I'd also rather not do opt-out as that will get unmanagable.

**Comment by Tratcher** [comment]
Can you do something in the Directory.Build.props for just those projects?

**Comment by jkotalik** [comment]
I don't think it would be that easy. If I add properties to subfolder directory.props, it would still affect non-sln builds. I'll tinker around with some ideas and see what I come up with.

**Comment by jkotalik** [comment]
Came up with a better solution (that I still don't like) now.

**Comment by jkotalik** [comment]
Getting this working in other Slns besides IISIntegration is more work. Right now, the slns for functional tests don't build native correctly (doesn't even attempt it), but still sees that the dlls aren't present. I'll address those in a separate PR as I'd like to unblock blazor.

File: src/Servers/IIS/build/assets.props [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <!-- This file is autogenerated -->
 <Project>
     <PropertyGroup>
+        <BuildIisNativeProjects Condition="'$(SolutionName)' == 'IISIntegration'">true</BuildIisNativeProjects>
```
Comment (Tratcher) [inline src/Servers/IIS/build/assets.props]: Which doesn't need to check we're running in VS because SolutionName is only ever set in VS?

Comment (jkotalik) [reply src/Servers/IIS/build/assets.props]: Yeah

**Review by Tratcher — APPROVED** [review]

**Pull request merged**